File: Examples/test-suite/perl5/using_private_runme.pl
use strict;
use warnings;
use Test::More tests => 5;
BEGIN { use_ok('using_private') }
require_ok('using_private');

my $f = using_private::FooBar->new();
$f->{x} = 3;

is($f->blah(4), 4, "blah(int)");

is($f->defaulted(), -1, "defaulted()");

is($f->defaulted(222), 222, "defaulted(222)");

File: Examples/test-suite/ocaml/using_private_runme.ml
open Swig
open Using_private

let _ =
  let f = new_FooBar '() in
  let _ = f -> "[x]" (3) in
  assert (f -> blah (4) as int = 4);
  assert (f -> defaulted () as int = -1);
  assert (f -> defaulted (222) as int = 222)
;;
